assets/helpers/paymentIntegrations/newPaymentFlow/readerRevenueApis.js

```diff
@@ -172,10 +172,21 @@ function postRegularPaymentRequest(
   setGuestAccountCreationToken: (string) => void,
   setThankYouPageStage: (ThankYouPageStage) => void,
 ): Promise<PaymentResult> {
-  return logPromise(fetchJson(
-    uri,
-    requestOptions(data, 'same-origin', 'POST', csrf),
-  ).then(checkRegularStatus(participations, csrf, setGuestAccountCreationToken, setThankYouPageStage)));
+  return logPromise(fetch(uri, requestOptions(data, 'same-origin', 'POST', csrf)))
+    .then((response) => {
+      if (response.status === 500) {
+        logException(`500 Error while trying to post to ${uri}`);
+        return ({ paymentStatus: 'failure', error: 'internal_error' });
```

Comment:
so to confirm my reading of the code - basically if we get a 500, we don't want to assume the content is anything meaningful (JSON), better to just replace whatever content we got back with this hard coded json and then we can deal with it properly later.
If this was Scala I'd want to return an Either where the Right is whatever actually came from the server, and if it's a left then we didn't get anything useful back.  Then we are not having to fake a response that didn't happen.  But I appreciate this is JS where turning everything into more simply typed json is more usual?

Reply:
I also wished for an Either 😭